### ISSUE TITLE
Unable to parse declarations with multiple `[`, `]` pairs

### ISSUE THREAD

**Issue by g5t** [McStasMcXtrace/mccode-antlr#87]
As found by @willend, example instruments which use the `Monochromator_bent` component do not pass code-generation with `mccode-antlr` due to a `RuntimError` complaining about 'multi-level' arrays.

Partial stack trace highlighting where the error originates
```cmd
... mccode_antlr/translators/c_listener.py", line 298, in visitDeclarator
    raise RuntimeError('No idea how to handle multi-level arrays')
```

This would appear to originate from the [definition of `double crystal_table[56][10]`](https://github.com/McStasMcXtrace/McCode/blob/9e5667820934d44c60f4466b07612f3d5cd22314/mcstas-comps/contrib/Monochromator_bent.comp#L284-L339)

It is probably more-sensible to not raise on this issue, but instead to handle it _somehow_.

**Comment by willend** [comment]
@g5t one (reasonable?) way to fix / work around this is likely to use a malloc'ed approach, in the component, i.e.

```
DArray2D crystal_table;
cystal_table=create_darray2d(56,10);
```
( Do you know if a simple assignment of the table values "in one go" `crystal_table = {{ ... }}` would work after the above declaration?) 

Plus then spread the word among comp writers that we prefer the use of the "array" definitions for 2/3 d arrays? (We have both are DArray and IArray as I remember...)

**Comment by willend** [comment]
Oh and btw prior to yesterdays related [PR om McCode](https://github.com/McStasMcXtrace/McCode/pull/1790) there were other code-bits that raised the same error without being (fully) related, e.g. this function declaration:

`double dSigdW(int sw, double Q, double Qtab[], double Itab[])`

(replacing by a standard `double* pointer` declaration worked just fine of course)

**Comment by g5t** [comment]
I've just merged #88 to handle the `double crystal_table[56][10]` case. 

The new issue you mentioned raised the same runtime error because there was only a check for more than a single `[`, `]` pair in one declaration statement.
With the changes I've made, that function (pre)declaration is parsed but the resulting information isn't entirely correct.
I doubt this will give problems until some unlikely distant future where I try to _use_ declared C functions by extracting their signature and definition from an instrument.

**Comment by willend** [comment]
Confirmed to function as expected, see https://new-nightly.mcstas.org/2024-12-07_output.html

Only two Union event-logger related instruments now don't compile (section of 2nd column from above table):
<img width="896" alt="Screenshot 2024-12-07 at 10 35 16" src="https://github.com/user-attachments/assets/b0d063d8-0855-46eb-996b-3d27431f2b18">
